### PR TITLE
chore: update the timerange

### LINKED ISSUE #5864: Updates to time range dropdown
* Can we make only the “Time range options” section scrollable and keep the rest of the dropdown a static?
* Since the timezone dropdown gets cut off I think we should make the bottom margin of the Time Zone dropdown 24px instead of 8px
* I spoke with Gary about this, but for the time range button we'd like to see what it would look like to make it dynamic and expand the button to show the full time range when a user has made that selection. Otherwise the button should remain the same width it is today. 
* Can we sync the calendar to match changes to the dates in the custom time range input fields?

### PR DESCRIPTION
**Pull request by asalem1**
Closes #5864

This PR updates some of the UX around the new time range dropdown. Most of the diff here is making the menu its own component. One of the reasons for this is that we were previously mounting the entire UI together, and so we didn't get reactivity benefits from the useEffect hook that we do know. 

### Checklist

Authors and Reviewer(s), please verify the following:

- [ ] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [ ] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [ ] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

**Review by drdelambre — DISMISSED** [review]

File: src/timeMachine/components/Vis.tsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ const TimeMachineVis: FC<Props> = ({
       !!giraffeResult.table.length)
 
   useEffect(() => {
-    if (viewProperties.hasOwnProperty('colors')) {
+    if (viewProperties.hasOwnProperty('colors') && !isViewingRawData) {
```
Comment (appletreeisyellow) [inline src/timeMachine/components/Vis.tsx:98]: Just curious, how is this related to time range?

File: src/shared/components/dateRangePicker/NewDatePicker.tsx [hunk 2]
```diff
@@ -220,9 +229,187 @@ const DatePicker: FC = () => {
...
+              <Input
+                className="date-picker__input"
+                onChange={handleSetStartDate}
+                status={
+                  inputStartErrorMessage === NBSP
+                    ? ComponentStatus.Default
+                    : ComponentStatus.Error
+                }
+                type={InputType.Text}
+                value={inputStartDate}
```
Comment (appletreeisyellow) [inline src/shared/components/dateRangePicker/NewDatePicker.tsx:298]:
```suggestion
                value={inputStartDate || ''}
```
Just an edge case, there is a dev warning when click on the time range picker before selecting any bucket. Adding an empty string should remove the dev warning

File: src/shared/components/dateRangePicker/NewDatePicker.tsx [hunk 3]
```diff
@@ -220,9 +229,187 @@ const DatePicker: FC = () => {
...
+              <Input
+                className="date-picker__input"
+                onChange={handleSetEndDate}
+                status={
+                  inputEndErrorMessage === NBSP
+                    ? ComponentStatus.Default
+                    : ComponentStatus.Error
+                }
+                type={InputType.Text}
+                value={inputEndDate}
```
Comment (appletreeisyellow) [inline src/shared/components/dateRangePicker/NewDatePicker.tsx:318]:
```suggestion
                value={inputEndDate || ''}
```
Same as above

File: src/shared/components/dateRangePicker/NewDatePicker.tsx [hunk 4]
```diff
@@ -138,14 +144,17 @@ const DatePicker: FC = () => {
       startInput = `${dateParts} ${timeParts}`
     }
     if (endInput instanceof Date) {
+      const endDate = new Date(endInput)
+      // this sets the time to the end of the selected end day
+      endDate.setMinutes(endDate.getMinutes() - 1)
```
Comment (appletreeisyellow) [inline src/shared/components/dateRangePicker/NewDatePicker.tsx:149]: Nice!

**Review by appletreeisyellow — COMMENTED** [review]
I left two comments related to dev warnings in an edge case. Not a big deal. It would be good to fix it in this PR so that the dev console is clean for future development

**Review by appletreeisyellow — APPROVED** [review]
